modules/nonce/tx_test.go

```diff
@@ -0,0 +1,88 @@
+package nonce
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/tendermint/basecoin"
+	"github.com/tendermint/basecoin/stack"
+	"github.com/tendermint/basecoin/state"
+)
+
+func TestNonce(t *testing.T) {
+	assert := assert.New(t)
+	require := require.New(t)
+
+	// generic args here...
+	chainID := "my-chain"
+	height := uint64(100)
+	ctx := stack.MockContext(chainID, height)
+	store := state.NewMemKVStore()
+
+	act1 := basecoin.Actor{App: "fooz", Address: []byte{1, 2, 3, 4}}
+	act2 := basecoin.Actor{App: "fooz", Address: []byte{1, 1, 1, 1}}
+	act3 := basecoin.Actor{App: "fooz", Address: []byte{3, 3, 3, 3}}
```

Comment:
same app, chain, different address is good... but can you add a few more with:
 same address, different app
 same app, address, different chain... (or prohibit signing with different chains here by adding that to the validation logic)

maybe you'll find some edge cases in `getSeqKey`....